config/initializers/content_security_policy.rb

```diff
@@ -32,16 +31,18 @@
   p.worker_src :self, :blob, assets_host
 
   if Rails.env.development?
-    webpacker_public_host = ENV.fetch('WEBPACKER_DEV_SERVER_PUBLIC', Webpacker.config.dev_server[:public])
-    front_end_build_urls = %w(ws http).map { |protocol| "#{protocol}#{Webpacker.dev_server.https? ? 's' : ''}://#{webpacker_public_host}" }
+    # Hacky solution to force CSP to correctly allow localhost, even if ViteRuby is bound to 0.0.0.0.
+    front_end_build_urls = %w(ws http).map { |protocol| "#{protocol}#{ViteRuby.config.https ? 's' : ''}://localhost:#{ViteRuby.config.port}" }
```

Comment:
there should be a better way to do this, but i couldn't quite read figure out how.